### DISCUSSION TITLE
Integration of WalletConnect into Chrome Extension

### DISCUSSION THREAD

**Discussion by heithemkacem** [WalletConnect/walletconnect-monorepo#3641] [Developer Support]
Title: Integration of WalletConnect into Chrome Extension

Message:
Hello, I have been working on integrating WalletConnect into my Chrome extension to allow users to connect to other Web3 websites seamlessly. My goal is to ensure the integration works flawlessly before submitting it.

Testing Integration:
I have completed the necessary setup in the dashboard but haven't submitted it yet. Now, I would like to test the integration. Is it possible to test WalletConnect integration locally?

Objectives:

Add WalletConnect functionality to the Chrome extension.
Enable users to connect to other Web3 websites using WalletConnect.
Display my wallet within the WalletConnect modal.
Testing Process:
I'd appreciate guidance on how to effectively test this integration and ensure it works as intended. Are there specific steps or best practices I should follow during testing?

Thank you in advance for your assistance.

**Comment by gintsmurans** [comment]
Hey!

I would guess that wallet connect is not going to work in extension because of manifest v2 which denies loading of remote scripts. Or did you already find a solution?

**Reply by heithemkacem** [reply]
hello @gintsmurans , the integration works as i have tried it with an open source project id , my question is can i add a custom wallet to the web3model with an RPC url and chain id ?

**Reply by heithemkacem** [reply]
i want to submit to wallet connect but they require test integration, so my question is how can i add my wallet to lab.web3model.com
